### PR TITLE
Small refactor and bump github.com/samber/lo

### PR DESCRIPTION
**Pull request by bryanhuhta**

### CODE REVIEW

**Comment by simonswine** [comment]
@bryanhuhta Nice work! I wonder if this will have any performance impact?

File: pkg/model/flamegraph.go [hunk 1]
```diff
@@ -29,7 +29,7 @@ func NewFlameGraph(t *Tree, maxNodes int64) *querierv1.FlameGraph {
 
 	minVal := t.minValue(maxNodes)
 
-	stack := stackNodePool.Get().(*Stack[stackNode])
+	stack := getNodeStackPool().Get().(*Stack[stackNode])
```
Comment (korniltsev) [inline pkg/model/flamegraph.go:32]: I dont get it. Does this create a fresh pool and fresh stack every time?

Comment (korniltsev) [reply pkg/model/flamegraph.go:32]: image

**Comment by simonswine** [comment]
@pyrobench BenchmarkFlamegraph

**Pull request closed without merging**